vimiv/api/_modules.py

```diff
@@ -89,6 +89,62 @@ def copy_name(abspath: bool = False, primary: bool = False) -> None:
     clipboard.setText(name, mode=mode)
 
 
+@api.keybindings.register("yi", "copy-image")
+@api.keybindings.register("yI", "copy-image --primary")
+@api.commands.register()
+def copy_image(
+    primary: bool = False,
+    width: int = None,
+    height: int = None,
+    size: int = None,
+    count: int = None,
+) -> None:
+    """Copy currently selected image to system clipboard.
+
+    **syntax:** ``:copy-image [--primary] [--width=WIDTH] [--height=HEIGHT] [--size=SIZE]``
+
+    optional arguments:
+        * ``--primary``: Copy to primary selection.
+        * ``--width``: Scale width to the specified value.
+        * ``--height``: Scale height to the specified value.
+        * ``--size``: Scale longer side to the specified value.
+
+    **count:** Equivalent to the ``--size``option
+    """
+    clipboard = QGuiApplication.clipboard()
+    mode = QClipboard.Selection if primary else QClipboard.Clipboard
+    path = api.current_path()
+
+    try:
+        reader = imagereader.get_reader(path)
+        pixmap = reader.get_pixmap()
+    except ValueError as e:
+        log.error(str(e))
+        return
+
+    if size or count:
+        pix_size = pixmap.size()
+
+        size = count if count else size
+
+        if pix_size.height() >= pix_size.width():
+            _logger.debug(f"Copy image with size {size} restricting height")
+            pixmap = pixmap.scaledToHeight(size)
```

Comment:
Feel free to add `  # type: ignore[arg-type]` to the two lines failing mypy. It probably cannot deal with this slightly non-trivial check to know that size is now certainly an `int` and no longer possibly `None`.

With this and splitting the syntax section into two lines
```
    **syntax:** ``:copy-image [--primary] [--width=WIDTH] [--height=HEIGHT]
    [--size=SIZE]``
```
the nasty CI should finally be happy (except for the annoying timeout issue) :laughing:

Reply:
All green, thanks for the hints!